src/pkrd/rng/mt.rs

```diff
@@ -0,0 +1,77 @@
+pub struct MT {
+    index: usize,
+    mt: [u32; 624],
+}
+
+impl MT {
+    pub fn new(seed: u32) -> MT {
+        let mut rng = MT {
+            mt: [0; 624],
+            index: 624,
+        };
+        rng.init(seed);
+
+        rng
+    }
+
+    fn init(&mut self, seed: u32) {
+        self.mt[0] = seed;
+
+        let mut seed = seed;
+        for i in 1..624 {
+            seed = 0x6c078965u32.wrapping_mul(seed ^ (seed >> 30)).wrapping_add(i);
+            self.mt[i as usize] = seed;
+        }
+    }
+
+    pub fn next(&mut self) -> u32 {
```

Comment:
What are thoughts on implementing rust's [Iterator trait](https://doc.rust-lang.org/std/iter/index.html)?

Implementing Iterator will give us free methods like `skip`, `find`, and `map`.  Then we can do things like `rng.skip(10).next()`, which seems like a common method we've implemented before.